### ISSUE TITLE
Revise and clean public API

### ISSUE THREAD

**Issue by FirefoxMetzger** [pygfx/pylinalg#67]
We want to revise and clean the public API as follows:

* Considering that the `pylinalg.obj` folder now only contains an empty init and that the "actual" object API lives in `pygfx.utils.transform`, we can remove the folder/submodule

* Drop `make_` in function names that include `from_`:

  To make function names a little less verbose, we discussed that it is possible to drop `make_` in functions that build objects `from_` some reference, e.g.

  - `quaternion_make_from_euler_angles` -> `quaternion_from_euler_angles`

  This loses no information while saving a view characters.

* We can make the API less verbose by introducing shorter function names and using namespaces to organize, e.g.:
  ```py
  cube.world.rotation = la.quat.mul(la.quat.from_euler((0.5, 0.01), order="XY"), cube.world.rotation)
  ```

**Comment by Korijn** [comment]
I merged #70 into this issue and added the concept for namespaces and shorter function names.

**Comment by almarklein** [comment]
That's a long list to digest, so let's start with a few general remarks.

I think we should be careful with abbreviations, because they add cognitive load (especially to people less familiar with the terms). The exception to this rule would be names that are commonly abbreviated (e.g. mat, vec) and perhaps abbreviations used in Numpy.

Not a fan of the `mxx`, `vxx` ones.

Would the namespacing mean submodules? An advantage of namespacing with a prefix is that something like `euclidian_to_spherical` would just fit in the main namespace without a "vec" prefix, I think?

> matrix_make_rotation_from_euler_angles
> 
>     matrix_make_rotation_from_euler_angles
>     mat.euler
>     mat.from_euler
>     meul

I really like `mat.from_euler` here. Would also be happy with `mat_from_euler`.

On that note, everywhere where we use the word "euler" or "axis_angle", we can probably omit the words "rotation" and "angle".

> `vector_euler_angles_from_quaternion`

What about `quaternion_to_euler`?
